custom_components/ha-google-home/api.py

```diff
@@ -93,6 +105,14 @@ async def get_alarms_and_timers(self, url, device):
         async with self._session.get(url, headers=HEADERS, timeout=TIMEOUT) as response:
             if response.status == HTTP_OK:
                 resp = await response.json()
+            elif response.status == HTTP_UNAUTHORIZED:
+                # If token is invalid - force reload homegraph providing new token and rerun the task
+                _LOGGER.debug(
+                    "Failed to fetch data from {device} due to invalid token. Will refresh the token and try again".format(
+                        device=device.device_name,
+                    )
+                )
+                self.google_devices = []
```

Comment:
This means we won't have updated results for this update cycle and will only get it after `UPDATE_INTERVAL`. It's better to retry immediately.

Reply:
Yeh, but tbh i don't know how to retrigger the entire coordinator update method from here. I kind of assumed waiting 10 sec (UPDATE INTERVAL) once a day might not be too big of a problem. Thou, if can see any good way to terminate the current look and create a new async task to update coordinator - that would be very helpful

Reply:
Let's keep it for other PR then.